File: cplusplus/design_pattern/VisitorProj/ObjectStructure.cpp
#include "stdafx.h"
#include "ObjectStructure.h"
#include "Node.h"
#include "Visitor.h"


ObjectStructure::ObjectStructure()
{
}


ObjectStructure::~ObjectStructure()
{
}


void ObjectStructure::addNode(std::shared_ptr<Node> node)
{
    this->collect.push_back(node);
}


void ObjectStructure::loopCollect(std::shared_ptr<Visitor> v)
{
    std::for_each(collect.begin(), collect.end(), [v](auto it) {it->accept(v); });
}

File: Java/pattern/VisitorProj/src/com/pattern/ObjectStructure.java
package com.pattern;
import java.util.ArrayList;

public class ObjectStructure {

	private ArrayList<Node> collect = new ArrayList<Node> (5);
	
	public void addNode(Node node) {
		collect.add(node);
	}
	
	public void loopCollect(Visitor v) {
		int len = collect.size();
		for (int i = 0; i < len; ++i) {
			collect.get(i).accept(v);
		}
	}
	
	
}
